### DISCUSSION TITLE
How data.TimePDE generates data for in the domain and boundary conditions?

### DISCUSSION THREAD

**Discussion by Aryahagh** [lululxvi/deepxde#678] [Q&A]
Hi there,
I'm using deepxde in a project and I have my own dataset. In this dataset I do have some particular values for each location and time. (in PDE(x,y) I do have both values for x and y). Now I'm training the model and observed in data.TimePDE, I can generate some amount of data in the domain (num_domain) and on the boundaries (num_boundaries). I just went through the deepxde documentations and understood these numbers generate some locations and time in the domain (so in PDE (x,y) it gives us some x) but we do not have value for those input data. When we do not have the actual values for those points how we are calculating loss when we predict some value for these locations. In the other word, how it can generate the solution even without any annotated dataset? 
Does my annotated dataset only acts as more data to train the model and do nothing else?

Thanks in advance for your help

**Comment by lululxvi** [comment]
- Please read the DeepXDE paper to understand how PINN works.
- If you have extra data, it would help, see FAQ "Q: How can I use a dataset of the solution?"
